modules/lib/generators.nix

```
@@ -1,5 +1,44 @@
 {lib}: {
-  # generatorName = {...}: {
-  #   function
-  # };
+  # These functions are inspired by home-manager's gtk generators.
+  gtk = let
+    inherit (builtins) isBool isString toString;
+    inherit (lib.strings) hasPrefix concatStringsSep;
+    inherit (lib.attrsets) mapAttrsToList;
+    inherit (lib.trivial) boolToString;
+    inherit (lib.generators) toINI;
+    inherit (lib.types) attrsOf oneOf bool int float str;
+  in {
+    gtkType = attrsOf (oneOf [
+      bool
+      int
+      float
+      str
+    ]);
+    toGtk2Text = let
+      formatGtk2 = n: v: let
+        n' = "gtk-" + n;
+        v' =
+          if isBool v
+          then boolToString v
+          else if isString v
+          then
+            if hasPrefix "GTK_" v
+            then v
+            else ''"${v}"''
+          else toString v;
```

I think we can get equal results in a cleaner manner by using attrsets to identify the type and act accordingly:
```suggestion
v' = {
  bool = boolToString v;
  string = if (hasPrefix "GTK_" v) then v else ''"${v}"'';
}.${builtins.typeOf v};
```
Granted, I don't know how many other types does gtk configurations support. If its much more than bool and string, it might be worth leaving the if-elses, since we need to have an attribute for each type this way.

---

I think the attrset way is mode readable, but since it also accepts floats and ints, we'd have to unnecessarily repeat the same statement 3 times :/ So i guess we can leave this section as it is.

---

I think the attrset is cleaner and easier to follow for people that aren't complete nix begginers. However this is really a nitpick at this point, so i'd say just let it be then :p

---

> It could also be an int or a float, so we'd have to add that. I think overall it's better just to use the if-then-else because in this case it makes the code more understandable.

I agree with this, it's not like gtk configuration's format supports a dozen types anyway.